src/index.ts

```diff
@@ -1,16 +1,36 @@
 import * as fs from "fs"
 import * as path from "path"
 import * as child_process from "child_process"
+import decompress from "decompress"
 
-const traverse = (dir: string) => {
+interface Package {
+    readonly name: string
+    readonly version: string
+}
+
+const traverse = async (dir: string): Promise<void> => {
     const files = fs.readdirSync(dir, { withFileTypes: true })
     for (const file of files) {
         const filePath = path.join(dir, file.name)
         if (file.isDirectory()) {
             traverse(filePath)
         } else if (path.extname(file.name) === ".tgz") {
-            console.log(`publishing ${filePath}`)
-            child_process.execSync(`npm publish ${filePath} --access=public`)
+            const files = await decompress(
+                filePath,
+                {
+                    filter: file => file.path === "package/package.json"
+                }
+            )
+            const packageJson = files[0]
+            const { name, version } = JSON.parse(packageJson.data.toString()) as Package
+            const id = `${name}@${version}`
+            const viewNames = child_process.execSync(`npm view ${id} name`).toString().split("\n")
```

Comment:
do we need to check the version of the package as well?
```bash
npm view ms-rest-js name version dist-tags
name = 'ms-rest-js'
version = '1.0.455'
dist-tags = { latest: '1.0.455', preview: '1.0.456' }
```
This would further help us verify whether the version to be published is already published or not

Reply:
That's interesting `preview` concept. I like it. For now, we will not use it but we may have another job which will switch from `preview` to `stable`.